src/nupic/bindings/algorithms.i

```diff
@@ -299,10 +299,18 @@ void forceRetentionOfImageSensorLiteLibrary(void) {
     self->load(inStream);
   }
 
-  inline void add_sample(float y_val, PyObject* x_vector)
+  inline PyObject* add_sample(float y_val, PyObject* x_vector)
   {
-    PyArrayObject* x = (PyArrayObject*) x_vector;
+    PyArrayObject* x = (PyArrayObject*)x_vector;
+    if (!PyArray_ISFLOAT(x) || PyArray_ITEMSIZE(x) != sizeof(float))
```

Comment:
@scottpurdy, if we want to verify that it's 32-bit, we should just compare `PyArray_ITEMSIZE(x)` to `4`; if we wish to verify that it's a single-precision float, then we should compare to `sizeof(float)`.

> float - single precision floating point type. Usually IEEE-754 32 bit floating point type

(http://en.cppreference.com/w/cpp/language/types)